CodeGeneration/Sources/SyntaxSupport/Child.swift

```diff
@@ -10,28 +10,79 @@
 //
 //===----------------------------------------------------------------------===//
 
+/// The kind of token a node can contain. Either a token of a specific kind or a
+/// keyword with the given text.
+public enum TokenChoice {
+  case keyword(text: String)
+  case token(tokenKind: String)
+
+  public var isKeyword: Bool {
+    switch self {
+    case .keyword: return true
+    case .token: return false
+    }
+  }
+}
+
+public enum ChildKind {
+  /// The child always contains a node of the given `kind`.
+  case node(kind: String)
+  /// The child always contains a child that matches one of the `choices`.
```

Comment:
```suggestion
  /// The child always contains a node that matches one of the `choices`.
```

But also, why separate this and the single kind case but not the single token vs tokens case?

Reply:
Because even if we have multiple `TokenChoice`s, the kind of the child is a `TokenSyntax`. But if there are multiple layout node choices, we need to create an enum for those choices. Also note that `nodeChoices` has `Child` (not a list of `String` the represent the different kinds) as the array kind, which means that `nodeChoices` would be a collection or list of tokens.